// hibp/dist/esm/api/haveibeenpwned/responses.mjs
/**
 * Known potential responses from the remote API.
 *
 * Unfortunately, the API does not send a decent human-readable message back
 * with each response, but they are documented on the website:
 * https://haveibeenpwned.com/api/v3#ResponseCodes
 *
 * These objects simply provide a mapping between the HTTP response status code
 * and the corresponding human-readable message so we can throw a more
 * descriptive error for the consumer. (They are also leveraged in our tests.)
 */
const emptyHeaders = /*#__PURE__*/new Map();
/** @internal */

const OK = {
  headers: emptyHeaders,
  status: 200
};
/** @internal */

const BAD_REQUEST = {
  headers: emptyHeaders,
  status: 400,
  statusText: 'Bad request — the account does not comply with an acceptable format.'
};
/**
 * This response has unique behavior. For some reason, the API includes an
 * object in the response body for this one, containing a human-readable
 * message. Manually populating the message here purely for use in tests.
 *
 * @internal
 */

const UNAUTHORIZED = {
  headers: emptyHeaders,
  status: 401,
  body: {
    statusCode: 401,
    message: 'Access denied due to missing hibp-api-key.'
  }
};
/** @internal */

const FORBIDDEN = {
  headers: emptyHeaders,
  status: 403,
  statusText: 'Forbidden - access denied.'
};
/** @internal */

const BLOCKED = {
  headers: /*#__PURE__*/new Map([['cf-ray', 'someRayId']]),
  status: 403
};
/** @internal */

const NOT_FOUND = {
  headers: emptyHeaders,
  status: 404
};
/**
 * This response has unique behavior. For some reason, the API includes an
 * object in the response body for this one, containing a human-readable
 * message. Manually populating the message here purely for use in tests.
 *
 * @internal
 */

const TOO_MANY_REQUESTS = {
  headers: emptyHeaders,
  status: 429,
  body: {
    statusCode: 429,
    message: 'Rate limit is exceeded. Try again in 2 seconds.'
  }
};
export { BAD_REQUEST, BLOCKED, FORBIDDEN, NOT_FOUND, OK, TOO_MANY_REQUESTS, UNAUTHORIZED };
